### ISSUE TITLE
Herald needs a maintainer!

### ISSUE THREAD

**Issue by Kestanous** [Meteor-Reaction/Herald#68]
Hi all, 

I will be brief. While love this project I have neither the time or ability to maintain it. The original app that started this died nearly a year ago. I have been trying to maintain it as I could but I think it would be best if someone else took the lead. The biggest issue I have is that I don't currently use it in any of my apps. 

If any is interested in taking over just helping out please make your self known here. Many thanks!

**Comment by Kestanous** [comment]
@dohomi, @alethes, @erasaur, @delgermurun, @splendido, @SachaG 
Call out to everyone on the contributor list, just want to be sure you see this, sorry for the spam.

**Comment by dohomi** [comment]
@Kestanous I'm very sorry to say, but for me Herald isn't in use for long time. I wouldn't be the right person to overtake.. I use Meteor only in combination with AngularJs.. Hope anyone else has time because the project has great potential.

**Comment by Kestanous** [comment]
Ah no worries @dohomi! Thanks for responding.

**Comment by SachaG** [comment]
I already have a lot on my plate, but that being said I'm still using Herald inside Telescope. So if someone ever wants to improve the notification system there they might be interested in becoming a maintainer of Herald?

**Comment by capensisma** [comment]
Hey Kestanous, You have an awesome project here!! love it!

Why can't just stay the maintainer ? just give 5 hours per week since the project is almost perfect ;)

I will try contributing on this awesome project!

Just don't drop the flag and keep up the good work!

**Comment by thatgibbyguy** [comment]
What's the roadmap for this and what are the demands?

**Comment by hellogerard** [comment]
Agree with @capensisma. You don't have to do a lot. Just keep up with the PRs and encourage them!

**Comment by splendido** [comment]
@Kestanous thanks for the proposal by I really have no time for this.
I'm sorry to hear you'd like someone else to take over your maintainer role...
Would it really be that much work for you to continue promoting contributions and keep an eye on it reviewing PRs?

**Comment by Kestanous** [comment]
To all, thanks for your support! For those that are asking why I can't keep supporting this. My issue is I am running 3 app and 3 other packages and don't even have the time for those (my company is still understaffed). If truly no one wants to take over then I will defiantly do what I can, but I worry about issues following though.

@thatgibbyguy For the most part this is stable. As for future work: 
* resolve the current issues, api / security / cleanup improvements.
* find a better server queuing system, the current one locks up. This is a big one, I can expand on this if you wish.
* A nice to have is more supported media (mobile push, SMS, ext.)
* Convert to ecmascript 2015 (make a Herald class its instances the couriers)

**Comment by thatgibbyguy** [comment]
@Kestanous I'd love to help, but this is probably a bit out of my scope of expertise. Give me another few months and I'll be up to speed :)

**Comment by hellogerard** [comment]
@thatgibbyguy It's a great way to learn!

**Comment by hellogerard** [comment]
@Kestanous Re: meteor-queue. I ran into a major issue with the queue as well. It was not clearing out old logs, so I collected over 1MM log records in my collection in a short amount of time. I submitted a PR fixing this issue (https://github.com/artwells/meteor-queue/pull/9), so if that was the culprit, meteor-queue should work much better now.

**Comment by victorperez** [comment]
@Kestanous I can help you with some issues specially for the Web Notification extension which I'm actually using in my project. I already submitted a PR (https://github.com/Meteor-Reaction/Herald-Web-Notifications/pull/9).

I'd like to help with the Herald package but actually I'm not familiar with the code, probably in a few weeks I will start using it in my project so by then I could give you a hand!

**Comment by thatgibbyguy** [comment]
Yeah I'm like @victorperez, I'm not totally familiar with this project, but it is a good one and needs help. Perhaps you can throw stuff to us, or we could moderate PRs?

**Comment by Kestanous** [comment]
Hey all. Hopping to get to this this weekend, time permitting. I am thinking that I can create a Herald Teem for my Meteor-Reaction and give push/pull rights to it. Then I could add any of you interested? Though it would probably be best if a single person took over coordinating things. Not sure how many people will end up joining but there is plenty of improvements to be done. 

As far as getting used to the code base goes. I feel like a full rewrite with the new ecmascript package might be in order. The current code has a bit of a spaghetti feel. I would be happy to lay out what Herald is supposed to do vs what it currently does. 

For now what are peoples thoughts? I should be able to make time this next month to get things set up properly for everyone. I might even be able to do said rewrite, though I make no promises when it comes to time :)

**Comment by capensisma** [comment]
Hello @Kestanous :)

I'm in ;) I will add some small features..
also can you tell us more about how do `What Herald is supposed to do vs what it currently does.`

We are ready to roll! :)
Thank you

**Comment by thatgibbyguy** [comment]
I can throw in some help on some tasks. It would be a great learning experience for me. Really into the team idea.

**Comment by Kestanous** [comment]
Hey all. Again sorry for the long wait. I just drafted my idea for herald v2.0 in this gist (untested code) https://gist.github.com/Kestanous/566914bd555fdccb82f1 

It removes a lot of bloat and allows the user to any logic they need for there app. (e.g.) no more built in background jobs but users can wrap the call in one. Thus fixing the queue problem completely. 

@SachaG I am particularly interested if you would use something like this with Telescope. Namely the fact that it depends on the ecmascript package. If you see any issue with it I would greatly appreciate knowing.

For everyone else. This code would reduce herald to a single file. One would think that would be easy enough to maintain :wink: I would likely ask people to take over the runners themselves as that is where a lot of the real logic would go. But a runner is just a function with some added logic given the type of media it uses.

I am thinking of creating an all new organization to hold the herald code. I am a huge fan of how things are set up with UserAccounts so likely something like that.

**Comment by SachaG** [comment]
I don't see a problem with using the Ecmascript package, looking forward to the new version :)
